A34-weighted-round-robin.md

```diff
@@ -1,103 +1,101 @@
+    // smaller `order_offset` will have higher priority. `order_offset` is assigned to this
+    // entry on constructing the priority queue for the first time and it's immutable.
+    // Also the `order_offset` assigned to entries is strictly increasing, in other words,
+    // no two entries have same `order_offset`.
     uint64 order_offset;
-    // `load_balancing_weight` of this endpoint from the latest `ClusterLoadAssignment`.
+
+    // `load_balancing_weight` of this endpoint from address attribute of this endpoint.
     double weight;
+
     // Subchannel data structure of this endpoint.
     Subchannel subchannel;
 }
 ```
+Initialization
+- At the very beginning, `deadline` of an entry `e` is equal to `1/e.weight`.
+- We assign `order_offset` to each entry while constructing the priority queue. `order_offset` assigned to an entry is distinct and nonnegative interger. During the whole lifecycle of this picker, `order_offset` of an entry is unchanged.
 
+Pick
 - On each call to the `Pick`, EDF picker picks the entry `e` on the top of the queue, returns the subchannel associated with the entry. After that, picker updates the `deadline` of `e` to `e.deadline + 1/weight` and either performs a pop and push the entry back to the queue or key increase operation.
-- `weighted_round_robin` updates the entries in EDF priority queue on any change of subchannel `ConnnectiveState` or endpoint `load_balancing_weight`.
-- If all endpoints have the same `load_balancing_weight`, `EDF` picker degenerates to `round_robin` picker. It's easier to reason and consistent with envoy.
-- Endpoints do not have `load_balancing_weight` assigned are discarded.
-- `weighted_round_robin` should always be updated to the lastest `ClusterLoadAssignment`. It's xDS server's responsibility to maintain consistency.
-
-#### EDF picker interface
-```
-/*
-Add new SubChannel to EDF picker.
 
-weighted_round_robin sees a new SubChannel in the most recent ClusterLoadAssignment or ConnectivityState of SubChannel is READY again.
-*/
-Add(SubChannel, weight)
+Notes
+- If all endpoints have the same `load_balancing_weight`, `EDF` picker degenerates to `round_robin` picker. The order of picked subschannel is purely decided by `order_offset`. It's easier to reason and consistent with envoy.
+- Endpoints do not have `load_balancing_weight` is assigned to 1 (the smallest possible weight). This is to be consistent with the [behavior of envoy on missing weight assignment](https://github.com/envoyproxy/envoy/blob/5d95032baa803f853e9120048b56c8be3dab4b0d/source/common/upstream/upstream_impl.cc#L359)
 
-/*
-Remove Subchannel from EDF picker.
-
-Subchannel is removed from the most recent ClusterLoadAssignment comparing to last ClusterLoadAssignment or
-ConnectivityState of SubChannel becomes not READY.
-*/
-Remove(SubChannel)
-
-/*
-Update the weight of SubChannel
+#### Service Config
+The service config for `weighted_round_robin` is very similar to `round_robin`
```

Comment:
I think this can just say something like "the config for the edf_weighted_round_robin LB policy is an empty proto message".